backend/app/api/ws/local_executor_namespace.py

```
@@ -0,0 +1,497 @@
+            )
+        except Exception as e:
+            logger.error(f"[LocalExecutor] Failed to update task status: {e}")
+            db.rollback()
+        finally:
+            db.close()
+
+    def get_connected_devices(self) -> list:
+        """Get list of connected devices."""
+        return list(self._devices.values())
+
+    async def dispatch_test_task(self, prompt: str, model_env: dict = None) -> dict:
```

Ruff (0.14.14)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)